graph/src/blockchain/mod.rs

```diff
@@ -81,13 +82,13 @@ pub trait Blockchain: Debug + Sized + Send + Sync + Unpin + 'static {
     type DataSource: DataSource<Self>;
     type UnresolvedDataSource: UnresolvedDataSource<Self>;
 
-    type DataSourceTemplate: DataSourceTemplate<Self>;
+    type DataSourceTemplate: DataSourceTemplate<Self> + Clone;
     type UnresolvedDataSourceTemplate: UnresolvedDataSourceTemplate<Self>;
 
     type TriggersAdapter: TriggersAdapter<Self>;
 
     /// Trigger data as parsed from the triggers adapter.
-    type TriggerData: TriggerData + Ord;
+    type TriggerData: TriggerData + Ord + Send + Sync + Clone;
```

Comment:
The `Clone` bound is not necessary.

Reply:
hmm it was needed for something but I can't remember what for. I'll remove it and see if everything still works 😛

Reply:
indeed seems like it's not needed, was a leftover from the previous version